### PR TITLE
Add file path quick-fix and autocomplete

### PR DESCRIPTION
**Pull request by JamesPHoughton**
## Summary
- **Quick-fix:** When a `file:` reference shows "File not found", suggests the closest matching workspace file (Levenshtein distance < 5) as a one-click fix
- **Autocomplete:** When typing after `file:`, suggests matching workspace files as completion items
- **Levenshtein utility:** Pure function with `findClosestMatch` helper, fully tested

### Security fixes from pre-commit review
- Cached file list with 5s TTL (avoids re-globbing on every cursor move)
- Glob metacharacter sanitization in autocomplete input
- File search scoped to relevant extensions (.prompt.md, .md, .yaml, etc.)

Refs #81

## Test plan
- [x] 16 new tests for levenshtein + findClosestMatch (edge cases: both empty, symmetry, multi-op, boundary threshold, exact match)
- [x] 120 total tests pass
- [x] Local security + spec reviews completed before commit
- [x] CI should pass (no VS Code API in test paths)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: apps/vscode/src/lib/levenshtein.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  for (const candidate of candidates) {
+    const distance = levenshtein(target, candidate);
+    if (distance < bestDistance) {
+      bestDistance = distance;
+      bestMatch = candidate;
+    }
+  }
+
+  return bestDistance < maxDistance ? bestMatch : null;
+}
```
[the remark this answers is not in the document]
Comment (JamesPHoughton) [reply apps/vscode/src/lib/levenshtein.ts:53]: Fixed — updated docstring to say 'strictly less than maxDistance'.

File: apps/vscode/src/extension.ts [hunk 2]
```diff
@@ -275,6 +276,130 @@ class TreatmentSemanticTokenProvider
...
+      );
+      this.cachedPaths = allFiles.map((f) =>
+        path.relative(workspaceFolder.uri.fsPath, f.fsPath),
+      );
+      this.cacheTimestamp = now;
+    }
+
+    for (const diagnostic of diagnostics) {
+      const badPath = diagnostic.message.replace("File not found: ", "");
+      const suggestion = findClosestMatch(badPath, this.cachedPaths);
```
[the remark this answers is not in the document]
Comment (JamesPHoughton) [reply apps/vscode/src/extension.ts:332]: Fixed — now uses `RelativePattern(workspaceFolder, ...)` to scope search to the document's workspace folder.

File: apps/vscode/src/extension.ts [hunk 3]
```diff
@@ -275,6 +276,130 @@ class TreatmentSemanticTokenProvider
...
+    // Refresh the cached file list if stale
+    const now = Date.now();
+    if (now - this.cacheTimestamp > FilePathQuickFixProvider.CACHE_TTL_MS) {
+      const allFiles = await vscode.workspace.findFiles(
+        "**/*.{prompt.md,md,yaml,jpg,jpeg,png,mp3,mp4}",
+        "**/node_modules/**",
+        5000,
+      );
+      this.cachedPaths = allFiles.map((f) =>
+        path.relative(workspaceFolder.uri.fsPath, f.fsPath),
```
[the remark this answers is not in the document]
Comment (JamesPHoughton) [reply apps/vscode/src/extension.ts]: Fixed — paths are normalized to POSIX separators via `.split(path.sep).join("/")`.

File: apps/vscode/src/extension.ts [hunk 4]
```diff
@@ -275,6 +276,130 @@ class TreatmentSemanticTokenProvider
...
+  async provideCompletionItems(
+    document: vscode.TextDocument,
+    position: vscode.Position,
+  ): Promise<vscode.CompletionItem[] | undefined> {
+    const line = document.lineAt(position).text;
+    const prefix = line.substring(0, position.character);
+
+    // Only trigger after "file:" with optional whitespace
+    if (!/^\s*file:\s/.test(prefix)) return undefined;
+
```
[the remark this answers is not in the document]
Comment (JamesPHoughton) [reply apps/vscode/src/extension.ts]: Fixed — relaxed regex from `^\\s*file:\\s` to `\\bfile:\\s` so it matches `- file:`, indented, and inline contexts.

File: apps/vscode/src/extension.ts [hunk 5]
```diff
@@ -275,6 +276,130 @@ class TreatmentSemanticTokenProvider
...
+      // Replace the entire value after "file: "
+      const valueStart = prefix.indexOf("file:") + "file:".length;
+      const whitespaceAfterColon =
+        prefix.substring(valueStart).match(/^\s*/)?.[0] ?? " ";
+      item.range = new vscode.Range(
+        position.line,
+        valueStart + whitespaceAfterColon.length,
+        position.line,
+        line.length,
+      );
```
[the remark this answers is not in the document]
Comment (JamesPHoughton) [reply apps/vscode/src/extension.ts]: Fixed — replacement range now stops at trailing YAML comments (`\\s+#`).

File: apps/vscode/src/extension.ts [hunk 6]
```diff
@@ -275,6 +276,130 @@ class TreatmentSemanticTokenProvider
...
+      globPattern,
+      "**/node_modules/**",
+      50,
+    );
+
+    return files.map((fileUri) => {
+      const relativePath = path.relative(
+        workspaceFolder.uri.fsPath,
+        fileUri.fsPath,
+      );
```
[the remark this answers is not in the document]
Comment (JamesPHoughton) [reply apps/vscode/src/extension.ts]: Fixed — autocomplete also uses `RelativePattern(workspaceFolder, ...)` now.
